src/plugins/navigation_embeddable/public/components/navigation_embeddable_link_editor.tsx

```diff
@@ -0,0 +1,167 @@
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useMemo, useState } from 'react';
+
+import {
+  EuiForm,
+  EuiIcon,
+  EuiButton,
+  EuiFormRow,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiRadioGroup,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiRadioGroupOption,
+  EuiButtonEmpty,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFocusTrap,
+  EuiFieldText,
+} from '@elastic/eui';
+import { css } from '@emotion/react';
+import { DashboardContainer } from '@kbn/dashboard-plugin/public/dashboard_container';
+// import { linksService } from '../../services/links_service';
+import { NavEmbeddableStrings } from './navigation_embeddable_strings';
+import {
+  DASHBOARD_LINK_TYPE,
+  EXTERNAL_LINK_TYPE,
+  NavigationEmbeddableLink,
+  NavigationLinkInfo,
+  NavigationLinkType,
+} from '../embeddable/types';
+import { DashboardLinkDestinationPicker } from './dashboard_link/dashboard_link_destination_picker';
+import { ExternalLinkDestinationPicker } from './external_link/external_link_destination_picker';
+
+export const NavigationEmbeddableLinkEditor = ({
+  onSave,
+  onClose,
+  parentDashboard,
+}: {
+  onClose: (closeBoth: boolean) => void;
+  onSave: (newLink: NavigationEmbeddableLink) => void;
+  parentDashboard?: DashboardContainer;
+}) => {
+  const [selectedLinkType, setSelectedLinkType] = useState<NavigationLinkType>(DASHBOARD_LINK_TYPE);
+  const [linkLabel, setLinkLabel] = useState<string>('');
+  const [linkDestination, setLinkDestination] = useState<string | undefined>();
+
+  const linkTypes: EuiRadioGroupOption[] = useMemo(() => {
+    return ([DASHBOARD_LINK_TYPE, EXTERNAL_LINK_TYPE] as NavigationLinkType[]).map((type) => {
+      return {
+        id: type,
+        label: (
+          <EuiFlexGroup
+            gutterSize="s"
+            alignItems="center"
+            aria-label={NavigationLinkInfo[type].description}
+          >
+            <EuiFlexItem grow={false}>
+              <EuiIcon type={NavigationLinkInfo[type].icon} color="text" />
+            </EuiFlexItem>
+            <EuiFlexItem>{NavigationLinkInfo[type].displayName}</EuiFlexItem>
+          </EuiFlexGroup>
+        ),
+      };
+    });
+  }, []);
+
+  return (
+    <EuiFocusTrap className={'navEmbeddableLinkEditor'}>
+      <EuiFlyoutHeader hasBorder>
+        <EuiButtonEmpty
+          css={css`
+            height: auto;
+          `}
+          flush="left"
+          color="text"
+          iconType={'arrowLeft'}
+          onClick={() => onClose(false)}
+        >
+          <EuiTitle size="m" aria-label="Go back to panel editor">
+            <h2>{NavEmbeddableStrings.editor.getAddButtonLabel()}</h2>
+          </EuiTitle>
+        </EuiButtonEmpty>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiForm component="form">
```

Comment:
Can we add validation to this form? https://elastic.github.io/eui/#/forms/form-validation

Reply:
I already have validation for the URL editor, like so:

![image](https://github.com/elastic/kibana/assets/8698078/643d233e-27e5-4610-976b-3719e7683781)

I don't really think anything in the dashboard link editor can be "invalid" - we just disable `Add link` until they, at the very least, select a dashboard:

![Jul-06-2023 16-08-42](https://github.com/elastic/kibana/assets/8698078/f2f38688-d3ff-407d-a32f-0bc208b2200a)

What do you think?

Reply:
My concern is that we only show the icon. Unless I'm missing something, we don't explain why the link is invalid. Your example is obviously invalid, but we also only allow `http:` and `https:` protocols. A `mailto:` protocol is also a valid URL, but we forbid it without explaining why.

Reply:
IMO it would be better to add that as part of https://github.com/elastic/kibana/issues/154381 - right now, my validation is a **very** basic RegEx match, and it possibly still allows for URLs/strings that we don't want. We should probably be using the same URL validation as the URL drilldown (I left a TODO comment about this so we won't forget), so I think once we have the real validation working, it will be a lot easier to tell the user why specifically their URL is invalid with a helpful error message :) 

Funny - I did a quick test and, even though the URL drilldown has some pretty robust URL validation in the background, we provide **no** error message even when adding a very obvious invalid URL.... We simply allow them to click "Create drilldown" but don't actually create the drilldown 😆 So we can definitely improve on that in our validation here

Reply:
I'm fine with pushing this to #154381.